src/App.tsx

```diff
@@ -1,9 +1,25 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.value === ' ') {
+      return event.target.value;
+    }
+
+    return setQuery(event.target.value);
+  };
+
+  const VisibleMovies = moviesFromServer.filter(movie => {
+    const queryLower = query.toLowerCase().trim();
+
+    return movie.title.toLowerCase().includes(queryLower)
```

Comment:
DRY
spit it with variables & create a separate function for the check if `str` includes `substr`
Also, you can make formating inside this function